src/pages/ReposSearchPage/ReposSearchPage.tsx

```diff
@@ -1,73 +1,119 @@
-import { useState } from 'react';
+import { createContext, useContext, useState } from 'react';
 import React from 'react';
 
 import Button from '@components/Button';
 import Input from '@components/Input';
 import RepoBranchesDrawer from '@components/RepoBranchesDrawer';
 import RepoTile from '@components/RepoTile';
 import SearchIcon from '@components/SearchIcon';
-import './ReposSearchPage.css';
 import GitHubStore from '@store/GitHubStore/GitHubStore';
 import { RepoItem } from '@store/GitHubStore/types';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import { Link, useHistory } from 'react-router-dom';
+
+import styles from './ReposSearchPage.module.scss';
+
+export type ReposContext = {
+  repoList: RepoItem[];
+  isLoading: boolean;
+  getOrgReposList: () => void;
+};
+
+const reposContext = createContext<ReposContext>({
+  repoList: [],
+  isLoading: false,
+  getOrgReposList: () => {},
+});
+
+const Provider = reposContext.Provider;
+
+export const useReposContext = () => useContext(reposContext);
 
 const ReposSearchPage = () => {
   const [enteredText, setEnteredText] = useState('ktsstudio');
   const [isLoading, setIsLoading] = useState(false);
   const [repoList, setRepoList] = useState<RepoItem[]>([]);
-  const [ChosenRepo, setChosenRepo] = useState<null | RepoItem>(null);
+  const [page, setPage] = useState(1);
+  const [hasMore, setHasMore] = useState(true);
 
-  const getOrgReposList = async (gitHubStore: GitHubStore) => {
+  const getOrgReposList = async () => {
     const result = await gitHubStore.getOrganizationReposList({
       organizationName: enteredText,
+      perPage: 10,
     });
     result.success ? setRepoList(result.data) : setRepoList([]);
   };
 
   const gitHubStore = new GitHubStore();
 
   React.useEffect(() => {
-    getOrgReposList(gitHubStore);
+    getOrgReposList();
   }, []);
 
   const onChange = (value: string) => {
     setEnteredText(value);
   };
 
+  const history = useHistory();
+
   const closeRepoBranchesDrawer = () => {
-    setChosenRepo(null);
+    history.push('/repos');
   };
 
   const buttonClick = (e: React.MouseEvent) => {
     setIsLoading(true);
-    getOrgReposList(gitHubStore);
+    getOrgReposList();
     return setIsLoading(false);
   };
 
+  const getReposListMore = async () => {
+    setPage((prevValue) => prevValue + 1);
+    const result = await gitHubStore.getOrganizationReposList({
+      organizationName: enteredText,
+      perPage: 20,
+      page: page,
+    });
+    if (result.success && result.data.length !== 0)
+      setRepoList((prevList) => prevList.concat(result.data));
```

Comment:
можно просто использовать `push`
```suggestion
      setRepoList((prevList) => prevList.push(...result.data));
```